js/app/modules/arrangement/sideBySide.js

```diff
@@ -51,66 +72,73 @@ const _SideBySideLayout = new Knowledge.Class({
-            return accum + card_nat;
-        }, 0);
+            nat += card_nat + SPACING_MAX;
+        });
 
-        nat += _HorizontalSpacing.XLARGE * (all_cards.length - 1);
-        return [min, nat];
+        return [min, nat - SPACING_MAX];
+    },
+
+    _ensure_card_parent: function (card, parent) {
+        let current_parent = card.get_parent();
+
+        if (current_parent != parent) {
+            current_parent.remove(card);
+            parent.add(card);
+            card.show();
+        }
     },
 
     vfunc_size_allocate: function (alloc) {
-        this.parent(alloc);
+        let all_cards = this.get_parent().get_cards();
 
         this.all_visible = true;
 
-        if (this.get_children().length === 0)
+        if (all_cards.length === 0)
             return;
 
-        let all_cards = this.get_parent().get_cards();
+        let [, button_nat] = this._popover_button.get_preferred_width();
+        let width = alloc.width;
+        let cards_width = 0;
+        let spacing = button_nat + SPACING_MIN;
 
-        let cards_width = all_cards.reduce((accum, card) => {
+        all_cards.forEach ((card) => {
             let [, card_nat] = card.get_preferred_width();
-            return accum + card_nat;
-        }, 0);
 
-        let leftover = (alloc.width - cards_width) / (all_cards.length - 1);
-        let spacing = this._get_horizontal_spacing(leftover);
-        let x = alloc.x;
-        let y = alloc.y;
+            cards_width += card_nat;
 
-        all_cards.forEach((card) => {
-            let [, card_nat] = card.get_preferred_width();
-            if (card_nat <= alloc.width) {
-                let offset = card_nat + spacing;
-                Arrangement.place_card(card, x, y, card_nat, MENU_HEIGHT);
-                alloc.width -= offset;
-                x += offset;
-            } else {
+            if (this.all_visible && width >= cards_width + spacing)
+                this._ensure_card_parent(card, this);
+            else  {
+                this._ensure_card_parent(card, this._popover_box);
                 this.all_visible = false;
-                card.set_child_visible(false);
             }
+
+            spacing += SPACING_MIN;
         });
-    },
 
-    _get_horizontal_spacing: function (width) {
-        let spacing;
-        if (width < _HorizontalSpacing.TINY) {
-            spacing = 0;
-        } else if (width < _HorizontalSpacing.SMALL) {
-            spacing = _HorizontalSpacing.TINY;
-        } else if (width < _HorizontalSpacing.LARGE) {
-            spacing = _HorizontalSpacing.SMALL;
-        } else if (width < _HorizontalSpacing.XLARGE) {
-            spacing = _HorizontalSpacing.LARGE;
-        } else {
-            spacing = _HorizontalSpacing.XLARGE;
+        /* Set proper spacing */
+        if (this.all_visible) {
+            let leftover = (width - cards_width) / (all_cards.length - 1);
+            this.spacing = Math.min(leftover, SPACING_MAX);
```

Comment:
optional: I think this could be improved by distributing 1 pixel from `(width - cards_width) % (all_cards.length - 1)` over the cards from start to finish